jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/UpdateChecker.java

```diff
@@ -0,0 +1,175 @@
+   * @param version the current version of Jib being used
+   * @param versionUrl the location to check for the latest version
+   * @param executorService the {@link ExecutorService}
+   * @return a new {@link UpdateChecker}
+   */
+  public static UpdateChecker checkForUpdate(
+      boolean skip, String version, String versionUrl, ExecutorService executorService) {
+    return new UpdateChecker(
+        executorService.submit(
+            () -> performUpdateCheck(skip, version, versionUrl, getConfigDir())));
+  }
+
+  @VisibleForTesting
+  static Optional<String> performUpdateCheck(
+      boolean skip, String currentVersion, String versionUrl, Path configDir) {
+    // Abort if offline or update checks are disabled
+    if (skip || Boolean.getBoolean(PropertyNames.DISABLE_UPDATE_CHECKS)) {
+      return Optional.empty();
+    }
+
+    try {
+      Path configFile = configDir.resolve("config.json");
+      if (Files.exists(configFile)) {
+        // Abort of update checks are disabled
+        ConfigJsonTemplate config =
+            JsonTemplateMapper.readJsonFromFile(configFile, ConfigJsonTemplate.class);
+        if (config.disableUpdateCheck) {
+          return Optional.empty();
+        }
+      } else {
+        // Generate config file if it doesn't exist
+        ConfigJsonTemplate config = new ConfigJsonTemplate();
+        Files.createDirectories(configDir);
+        JsonTemplateMapper.writeTo(config, Files.newOutputStream(configFile));
+      }
+
+      // Check time of last update check
+      Path lastUpdateCheck = configDir.resolve("lastUpdateCheck");
+      if (Files.exists(lastUpdateCheck)) {
+        String fileContents =
+            new String(Files.readAllBytes(lastUpdateCheck), StandardCharsets.UTF_8);
+        try {
+          Instant modifiedTime = Instant.parse(fileContents);
+          if (modifiedTime.plus(Duration.ofDays(1)).isAfter(Instant.now())) {
+            return Optional.empty();
+          }
+        } catch (DateTimeParseException ex) {
+          // Ignore parse failure; assume update check hasn't been performed
+        }
+      }
+
+      // Check for update
+      URLConnection connection = new URL(versionUrl).openConnection();
+      BufferedReader bufferedReader =
+          new BufferedReader(
+              new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
+      String latestVersion = bufferedReader.readLine().trim();
+      Files.write(lastUpdateCheck, Instant.now().toString().getBytes(StandardCharsets.UTF_8));
+      if (currentVersion.equals(latestVersion)) {
+        return Optional.empty();
+      }
+      return Optional.of(
+          "A new version of Jib ("
+              + latestVersion
+              + ") is available (currently using "
+              + currentVersion
+              + "). Update your build configuration to use the latest features and fixes!");
+
+    } catch (IOException ignored) {
+      // Fail silently
```

Comment:
How about attempting to delete both the config file and `lastUpdateCheck`? That can fix the situation if any of the files is corrupt.